core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,468 @@
+                  thirdCollectionNode]);
+      expect(CollectionLinearizerService.removeCollectionNode(collection,
+          'exp_id0')).toBe(true);
+      expect(collection.containsCollectionNode('exp_id0')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([secondCollectionNode, thirdCollectionNode]);
+    });
+
+    it('should correctly remove the last node from a collection', function() {
+      var collection = createLinearCollection();
+      expect(collection.containsCollectionNode('exp_id2')).toBe(true);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+      expect(CollectionLinearizerService.removeCollectionNode(collection,
+          'exp_id2')).toBe(true);
+      expect(collection.containsCollectionNode('exp_id2')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode]);
+    });
+
+    it('should correctly remove a middle node from a collection', function() {
+      var collection = createLinearCollection();
+      expect(collection.containsCollectionNode('exp_id1')).toBe(true);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+      expect(CollectionLinearizerService.removeCollectionNode(collection,
+          'exp_id1')).toBe(true);
+      expect(collection.containsCollectionNode('exp_id1')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, thirdCollectionNode]);
+    });
+  });
+
+  describe('appendCollectionNode()', function() {
+    it('should correctly append a node to an empty collection', function() {
+      var collection = CollectionObjectFactory.createEmptyCollection();
+      expect(collection.containsCollectionNode('exp_id0')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+                  collection)).toEqual([]);
+      expect(firstCollectionNode.getAcquiredSkillList().getSkills()).toEqual(
+          []);
+      CollectionLinearizerService.appendCollectionNode(collection, 'exp_id0',
+          firstCollectionNode.getExplorationSummaryObject());
+      firstCollectionNode = collection.getCollectionNodeByExplorationId(
+          'exp_id0');
+      expect(
+          firstCollectionNode.getPrerequisiteSkillList().getSkills()).toEqual(
+              []);
+      expect(firstCollectionNode.getAcquiredSkillList().getSkills()).toEqual([
+          'exp title0']);
```

Comment:
I'm not convinced testing prerequisite/acquired skills is beneficial. I feel like it's going to lead to fragile tests. It's also testing an implementation detail, rather than an interface detail. Suggest removing the prerequisite/acquired expects statements from this and other tests, unless the contract of one of the collection linearizer service methods specifically indicates it changes the skill lists.